File: backend/src/main/java/com/healthai/model/response/ApiResponse.java
package com.healthai.model.response;

import lombok.Data;

@Data
public class ApiResponse<T> {

    private int code;
    private String message;
    private T data;

    // Success response constructor
    public ApiResponse(T data) {
        this.code = 0;
        this.message = "success";
        this.data = data;
    }

    // Error response constructor
    public ApiResponse(int code, String message) {
        this.code = code;
        this.message = message;
        this.data = null;
    }

    // Static factory method for success response
    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(data);
    }

    // Static factory method for error response
    public static <T> ApiResponse<T> error(int code, String message) {
        return new ApiResponse<>(code, message);
    }
}

File: frontend/android/app/src/main/java/com/healthai/assistant/model/ApiResponse.kt
package com.healthai.assistant.model

/**
 * API响应基类
 */
data class ApiResponse<T>(
    val code: Int,
    val message: String,
    val data: T? = null
)

/**
 * 基础响应类
 */
data class BaseResponse(
    val success: Boolean
)

/**
 * 发送验证码响应
 */
data class SendCodeResponse(
    val sendTime: Long
)

/**
 * 登录响应
 */
data class LoginResponse(
    val token: String,
    val userId: String
)

/**
 * 发送验证码请求
 */
data class SendCodeRequest(
    val phone: String
)

/**
 * 登录请求
 */
data class LoginRequest(
    val phone: String,
    val code: String
)

/**
 * 激活用户请求
 */
data class ActivateUserRequest(
    val userId: String,
    val token: String
)
